File: ViewController.m
//
//  ViewController.m
//  Singleton(ARC)
//
//  Created by gan on 16/6/1.
//  Copyright © 2016年 GZLProject. All rights reserved.
//

#import "ViewController.h"
#import "Singleton.h"
@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    UIImageView * img = [[UIImageView alloc]initWithFrame:CGRectMake(0, 20,self.view.frame.size.width, self.view.frame.size.height-20)];
//    img.backgroundColor = [UIColor yellowColor];
    img.image = [UIImage imageNamed:@"IOS实现单例模式的几种实现方法.png"];
    [self.view addSubview:img];
//    Singleton *A = [[Singleton alloc] init];
//    
//    NSLog(@"A:%@",A);
    
    Singleton *B = [Singleton sharedInstance];
    
    NSLog(@"B:%@",B);
    
//    Singleton *C = [A copy];
//    
//    NSLog(@"C:%@",C);
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Swift练手2/Swift练手2/ViewController.swift
//
//  ViewController.swift
//  Swift练手2
//
//  Created by Asxce, on 16/4/6.
//  Copyright © 2016年 Asxce,. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
      
        class Shape{
            var numberOfSides = 0
            func simpleDescription() -> String {
                return"A shape with\(numberOfSides)sides"
            }
            
        }
        
        let shape = Shape()
        shape.numberOfSides = 7
        let shapeDescription = shape.simpleDescription()
        print("调用Shape类里面的函数",shapeDescription)
        //这是测试
        print("")
        
    
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

